tck/src/main/java/org/eclipse/microprofile/lra/tck/TckTests.java

```diff
@@ -45,398 +36,508 @@
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestName;
+import org.junit.runner.RunWith;
 
+@RunWith(Arquillian.class)
 public class TckTests {
-    private static final Long LRA_TIMEOUT_MILLIS = 50000L;
-    private static URL micrserviceBaseUrl;
-    private static URL rcBaseUrl;
+    private static final Logger LOGGER = Logger.getLogger(TckTests.class.getName());
+    private final Long LRA_TIMEOUT_MILLIS = 50000L;
+
+    /**
+     * <p>
+     * Timeout factor which adjusts waiting time and timeouts for the TCK suite.
+     * <p>
+     * The default value is set to <code>1.0</code> which means the defined timeout
+     * is multiplied by <code>1</code>.
+     * <p>
+     * If you wish the test waits longer then set the value bigger than <code>1.0</code>.
+     * If you wish the test waits shorter time than designed
+     * or the timeout is elapsed faster then set the value less than <code>1.0</code> 
+     */
+    @Inject @ConfigProperty(name = "lra.tck.timeout.factor", defaultValue = "1.0")
+    private double timeoutFactor;
+    
+    /**
+     * Number of seconds the TCK suite waits for coordinator to be available.
+     * The coordinator will is expected to be available at the hostname and port
+     * defined by config properties {@link #coordinatorHostName} and {@link #coordinatorPort}. 
+     */
+    @Inject @ConfigProperty(name = "lra.tck.coordinator.waiting", defaultValue = "5")
+    private int coordinatorWaitingTime;
+    
+    /**
+     * Host name where LRA coordinator is present.
+     */
+    @Inject @ConfigProperty(name = LRAClient.LRA_COORDINATOR_HOST_KEY, defaultValue = "localhost")
+    private String coordinatorHostName;
+    
+    /**
+     * Port where LRA coordinator is present.
+     */
+    @Inject @ConfigProperty(name = LRAClient.LRA_COORDINATOR_PORT_KEY, defaultValue = "8080")
+    private int coordinatorPort;
+
+    /**
+     * Host name where LRA recovery is expected to be launch and TCK suite tries to connect to it at.
+     * The port is specifed by {@link #recoveryPort}.
+     */
+    @Inject @ConfigProperty(name = LRAClient.LRA_RECOVERY_HOST_KEY, defaultValue = "localhost")
+    private String recoveryHostName;
+    
+    /**
+     * Port where LRA recovery is expected to be launch and TCK suite tries to connect to it at.
+     * The hostname is specifed by {@link #recoveryHostName}.
+     */
+    @Inject @ConfigProperty(name = LRAClient.LRA_RECOVERY_PORT_KEY, defaultValue = "8080")
+    private int recoveryPort;
+    
+    /**
+     * Path where recovery is available to accept requests.
+     * The hostname of LRA recovery is specifed by {@link #recoveryHostName},
+     * the port of LRA recovery is defined by {@link #recoveryPort}.
+     */
+    @Inject @ConfigProperty(name = LRAClient.LRA_RECOVERY_PATH_KEY, defaultValue = "lra-recovery-coordinator")
+    private int recoveryPath;
+
```

Comment:
PR #68 is removing all of these properties

Reply:
ok, I see. There will be just `lra.http.recovery.url`. When the PR#68 is merged I can change this for url too.